dev_tools/utils/nxdl_utils.py

```diff
@@ -357,14 +326,9 @@ def find_definition_file(bc_name):
     """
     bc_filename = None
     for nxdl_folder in ["contributed_definitions", "base_classes", "applications"]:
-        if os.path.exists(
-            f"{get_nexus_definitions_path()}{os.sep}"
-            f"{nxdl_folder}{os.sep}{bc_name}.nxdl.xml"
-        ):
-            bc_filename = (
-                f"{get_nexus_definitions_path()}{os.sep}"
-                f"{nxdl_folder}{os.sep}{bc_name}.nxdl.xml"
-            )
+        nxdl_file = nexus_def_path / nxdl_folder / f"{bc_name}.nxdl.xml"
```

Comment:
existence of an f-string is not the same as existent of an instance of object PurePath

Reply:
yes, it is tested below